infra/aws/terraform/cncf-k8s-infra-aws-capa-ami/oidc.tf

```diff
@@ -0,0 +1,42 @@
+resource "aws_iam_openid_connect_provider" "github" {
+  url = "https://token.actions.githubusercontent.com"
+
+  client_id_list = [
+    "sts.amazonaws.com",
+  ]
+
+  thumbprint_list = ["d89e3bd43d5d909b47a18977aa9d5ce36cee184c"]
+}
+
+data "aws_iam_policy_document" "capa-gh-trust" {
+    statement {
+      actions = ["sts:AssumeRoleWithWebIdentity"]
+      
+      principals {
+        type        = "Federated"
+        identifiers = [aws_iam_openid_connect_provider.github.arn]
+      }
+
+      condition {
+        test = "ForAnyValue:StringEquals"
+        variable = "token.actions.githubusercontent.com:aud"
+        values = ["sts.amazonaws.com"]
+      }
+
+      condition {
+        test = "StringLike"
+        variable = "token.actions.githubusercontent.com:sub"
+        values = ["repo:kubernetes-sigs/cluster-api-provider-aws"]
+      }
+    }
+}
+
+resource "aws_iam_role" "gh-image-builder" {
```

Comment:
and likewise, we do have this https://github.com/terraform-aws-modules/terraform-aws-iam/tree/master/modules/iam-github-oidc-role which should work with both standard and enterprise

Reply:
this role would be specific to the use case so it would go here, unlike the provider which its location might be TBD

Reply:
Updated to use this module as well.